### PR TITLE
docs: mark two params as gov modifiable

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/celestia-app/issues/2972

### CODE REVIEW

File: specs/src/specs/params.md [hunk 1]
```diff
@@ -34,7 +34,7 @@ are blocked by the `x/paramfilter` module.
...
 | consensus.validator.PubKeyTypes               | Ed25519                                     | The type of public key used by validators.                                                                                                                                                      | False                     |
-| consensus.Version.AppVersion                  | 1                                           | Determines protocol rules used for a given height. Incremented by the application upon an upgrade.                                                                                              | False                     |
+| consensus.Version.AppVersion                  | 1                                           | Determines protocol rules used for a given height. Incremented by the application upon an upgrade.                                                                                              | True                      |
```
Comment (cmwaters) [inline specs/src/specs/params.md:37]: This isn't really modifiable by governance. The upgrade protocol is separate to governance

Comment (rootulp) [reply specs/src/specs/params.md:37]: hmm the tests made me think that this was gov modifiable but I'll try on a local devnet. It's possible the tests environment doesn't reflect reality and something else prevents this param from getting updated via governance.

Comment (cmwaters) [reply specs/src/specs/params.md:37]: Ok well we don't want governance to be able to change the app version.

Do you want to check the RPC endpoint `ConsensusParams` to see if it changed or also check the version of the header?

Comment (rootulp) [reply specs/src/specs/params.md:37]: Yes both great ideas, thanks!

🤯 it actually modified the app version

```
$ curl -X GET http://0.0.0.0:26657/consensus_params  | jq .
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   318  100   318    0     0  99499      0 --:--:-- --:--:-- --:--:--  310k
{
  "jsonrpc": "2.0",
  "id": -1,
  "result": {
    "block_height": "34",
    "consensus_params": {
      "block": {
        "max_bytes": "1974272",
        "max_gas": "-1",
        "time_iota_ms": "1"
      },
      "evidence": {
        "max_age_num_blocks": "120961",
        "max_age_duration": "1814400000000000",
        "max_bytes": "1048576"
      },
      "validator": {
        "pub_key_types": [
          "ed25519"
        ]
      },
      "version": {
        "app_version": "3"
      }
    }
  }
}
```

and in the block header

```
$ curl -X GET http://0.0.0.0:26657/block?height=36  | jq .
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1792  100  1792    0     0   437k      0 --:--:-- --:--:-- --:--:-- 1750k
{
  "jsonrpc": "2.0",
  "id": -1,
  "result": {
    "block_id": {
      "hash": "ED84FD959FCC69CF943113EFE1A0AF29B23C179C5AE252980840A59F9BFA2CA5",
      "parts": {
        "total": 1,
        "hash": "EEB79C3A23EC93B8AB4BC9B414E457CEFE5C72D7C8872D70F6D3C8AD70E2CBEF"
      }
    },
    "block": {
      "header": {
        "version": {
          "block": "11",
          "app": "3"
        },
        "chain_id": "private",
        "height": "36",
        "time": "2024-01-18T21:41:18.676555Z",
        "last_block_id": {
          "hash": "CE19AEE5447C0FFC5064861F602C192A087662D2443D58B028C40118714ECB7F",
          "parts": {
            "total": 1,
            "hash": "7CD7A0490ED1616028D7D1C0E08D0700D19F45DDA8406E2D3A1ABB06B89A47CB"
          }
        },
        "last_commit_hash": "3950A7ACAAD71F29CCE0D048997D7FA6FE42CC7D5A86073AA6318BC4CCD258DC",
        "data_hash": "3D96B7D238E7E0456F6AF8E7CDF0A67BD6CF9C2089ECB559C659DCAA1F880353",
        "validators_hash": "12811879BCAF510B19DF19DDC66E12FA2B71E76B7426FE29D36D695AE847B795",
        "next_validators_hash": "12811879BCAF510B19DF19DDC66E12FA2B71E76B7426FE29D36D695AE847B795",
        "consensus_hash": "C0B6A634B72AE9687EA53B6D277A73ABA1386BA3CFC6D0F26963602F7F6FFCD6",
        "app_hash": "D29EC9DCE455B31A939BB36366FB7BECA65B62ACB7596AA85EF907AB3D9DA5CD",
        "last_results_hash": "E3B0C44298FC1C149AFBF4C8996FB92427AE41E4649B934CA495991B7852B855",
        "evidence_hash": "E3B0C44298FC1C149AFBF4C8996FB92427AE41E4649B934CA495991B7852B855",
        "proposer_address": "64D9A517E6B8A1488B4C0D20CA3593B94C8DA87F"
      },
      "data": {
        "txs": [],
        "square_size": "1"
      },
      "evidence": {
        "evidence": []
      },
      "last_commit": {
        "height": "35",
        "round": 0,
        "block_id": {
          "hash": "CE19AEE5447C0FFC5064861F602C192A087662D2443D58B028C40118714ECB7F",
          "parts": {
            "total": 1,
            "hash": "7CD7A0490ED1616028D7D1C0E08D0700D19F45DDA8406E2D3A1ABB06B89A47CB"
          }
        },
        "signatures": [
          {
            "block_id_flag": 2,
            "validator_address": "64D9A517E6B8A1488B4C0D20CA3593B94C8DA87F",
            "timestamp": "2024-01-18T21:41:18.676555Z",
            "signature": "225ZL2Ve2IRRNCk5fYEO2zcNcdk/LkNdZmoe7QNqQhuuVejcPHNs28eOHbamkiV6/54rYTrqBA4BDjp+RxVkDA=="
          }
        ]
      }
    }
  }
}
```

Comment (rootulp) [reply specs/src/specs/params.md:37]: > Ok well we don't want governance to be able to change the app version.

Yea agreed. But this PR corrects the specs. In v2 we can make this param non gov modifiable.

Comment (cmwaters) [reply specs/src/specs/params.md:37]: Yup let's do that

Comment (cmwaters) [reply specs/src/specs/params.md:37]: Good find!

**Review by evan-forbes — APPROVED** [review]

File: specs/src/specs/params.md (hunk 1 above)
Comment (evan-forbes) [reply specs/src/specs/params.md:37]: oh shit yeah nice find

**Comment by evan-forbes** [comment]
what do we think about keeping the param as not modifiable in the specs since we're planning on fixing it for v2?

**Comment by rootulp** [comment]
> what do we think about keeping the param as not modifiable in the specs since we're planning on fixing it for v2?

I agree we should make this non gov modifiable in v2 but I think the specs should reflect what exists in the protocol rather than what we want. Note: we already fixed this in the living document CIP: https://github.com/celestiaorg/CIPs/pull/43

**Review by ninabarbakadze — APPROVED** [review]

**Pull request merged**